packages/datadog-plugin-jest-environment-node/src/index.js

```diff
@@ -0,0 +1,180 @@
+      await new Promise((resolve) => {
+        tracer._exporter._writer.flush(resolve)
+      })
+      return teardown.call(this)
+    }
+  }
+}
+
+function createHandleTestEvent (tracer, extraMetadata) {
+  return async function (event) {
+    if (event.name === 'test_skip' || event.name === 'test_todo') {
+      const childOf = tracer.extract('text_map', {
+        'x-datadog-trace-id': id().toString(10),
+        'x-datadog-parent-id': '0000000000000000',
+        'x-datadog-sampled': 1
+      })
+      tracer.startSpan(
+        'jest.test',
+        {
+          childOf,
+          tags: {
+            [SPAN_TYPE]: 'test',
+            [RESOURCE_NAME]: `${event.test.parent.name}.${event.test.name}`,
+            [TEST_TYPE]: 'test',
+            [TEST_NAME]: event.test.name,
+            [TEST_SUITE]: this.testSuite,
+            [TEST_STATUS]: 'skip',
+            [SAMPLING_RULE_DECISION]: 1,
+            ...extraMetadata
+          }
+        }
+      ).finish()
+    }
+    if (event.name === 'test_start') {
+      const childOf = tracer.extract('text_map', {
+        'x-datadog-trace-id': id().toString(10),
+        'x-datadog-parent-id': '0000000000000000',
+        'x-datadog-sampled': 1
+      })
+      let specFunction = event.test.fn
+      if (specFunction.length) {
+        specFunction = promisify(specFunction)
+      }
+      event.test.fn = tracer.wrap(
+        'jest.test',
+        { type: 'test',
+          childOf,
+          resource: `${event.test.parent.name}.${event.test.name}`,
+          tags: {
+            [TEST_TYPE]: 'test',
+            [TEST_NAME]: event.test.name,
+            [TEST_SUITE]: this.testSuite,
+            [SAMPLING_RULE_DECISION]: 1,
+            ...extraMetadata
+          } },
+        async () => {
+          let result
+          try {
+            result = await specFunction()
+            tracer.scope().active().setTag(TEST_STATUS, 'pass')
+          } catch (error) {
+            tracer.scope().active().setTag(TEST_STATUS, 'fail')
+            throw error
+          } finally {
+            tracer
+              .scope()
+              .active()
+              .context()._trace.started.forEach((span) => {
+                span.finish()
+              })
+          }
+          return result
+        }
+      )
+    }
+  }
+}
+
+module.exports = {
+  name: 'jest-environment-node',
+  versions: ['>=24.8.0'],
+  patch: function (NodeEnvironment, tracer) {
+    // TODO: eventually these will come from the tracer (generally available)
+    const ciMetadata = getCIMetadata()
+    const gitMetadata = getGitMetadata()
+    const envMetadata = getEnvMetadata()
+
+    const extraMetadata = {
+      [TEST_FRAMEWORK]: 'jest',
+      ...ciMetadata,
+      ...gitMetadata,
+      ...envMetadata
+    }
+    this.wrap(NodeEnvironment.prototype, 'teardown', createWrapTeardown(tracer))
+    NodeEnvironment.prototype.handleTestEvent = createHandleTestEvent(tracer, extraMetadata)
```

Comment:
this is not by default on `NodeEnvironment.prototype`. `jest` checks if it exists in https://github.com/facebook/jest/blob/c8418dfab714ddd75dcaaca28f9811c7da3e6123/packages/jest-circus/src/legacy-code-todo-rewrite/jestAdapterInit.ts#L121. Therefore I can't `wrap` it (it fails). I do delete the method in the `unpatch` though